lib/shopify-cli/app_type_command.rb

```diff
@@ -0,0 +1,25 @@
+require 'shopify_cli'
+
+module ShopifyCli
+  class AppTypeCommand < ShopifyCli::Command
+    class << self
+      def type(identifier, const, path)
+        autoload(const, "shopify-cli/commands/#{path}") if path
+        lookup[self] = {} unless lookup[self]
+        lookup[self][identifier] = const_get(const)
+      end
+
+      def lookup
+        @lookup ||= {}
+      end
+    end
+
+    def call(*args)
+      project = Project.current
+      cmd = self.class.lookup[self.class][project.app_type_id].new
```

Comment:
I ran into a problem when a command isn't supported for an app type.  I put this here to check, and display "unsupported in this context message":  https://github.com/Shopify/temp-shopify-app-cli/blob/da57151fbf80b5d118c4feb760ce0e21c7f11729/lib/shopify-cli/app_type_command.rb#L21

You probably know a better way of handling such error.

Reply:
Great idea, I'll add that to this PR.